src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h

```diff
@@ -45,51 +43,95 @@
 #include <OpenMS/DATASTRUCTURES/ListUtils.h>
 #include <OpenMS/METADATA/MetaInfoInterface.h>
 
+#include <xercesc/util/XMLString.hpp>
 #include <xercesc/sax2/DefaultHandler.hpp>
 #include <xercesc/sax/Locator.hpp>
 #include <xercesc/sax2/Attributes.hpp>
 
 #include <algorithm>
+#include <iosfwd>
+#include <string>
 
 namespace OpenMS
 {
   namespace Internal
   {
 
     /*
-     * @brief Helper class for XML parsing that handles the memory management for conversions of Xerces strings
+     * @brief Helper class for XML parsing that handles the conversions of Xerces strings
      *
      * It provides the convert() function which internally calls
      * XMLString::transcode and ensures that the memory is released properly
-     * through XMLString::release in a transparent and object oriented manner. 
-     *
-     * Releasing the memory can be done manually through clear() or will be done
-     * automatically in the destructor. 
+     * through XMLString::release internally. It returns a std::string or
+     * std::basic_string<XMLCh> to the caller who takes ownership of the data.
      *
     */
     class OPENMS_DLLAPI StringManager
     {
+
+      typedef std::basic_string<XMLCh> XercesString;
+
+      // Converts from a narrow-character string to a wide-character string.
+      inline XercesString fromNative(const char* str) const
```

Comment:
well... technically, these should be fromNative_, to adhere to OpenMS coding conventions...
/nitpick